Jint.Tests/Runtime/ModuleTests.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using Jint.Native;
+using Jint.Runtime;
+using Xunit;
+
+namespace Jint.Tests.Runtime
+{
+    public class ModuleTests
+    {
+        private readonly Engine _engine;
+
+        public ModuleTests()
+        {
+            _engine = new Engine();
+        }
+
+        [Fact]
+        public void ShouldExportNamed()
+        {
+            _engine.DefineModule(@"export const value = 'exported value';", "my-module");
+            var ns = _engine.ImportModule("my-module");
+
+            Assert.Equal("exported value", ns.Get("value").AsString());
+        }
+
+        [Fact]
+        public void ShouldExportNamedListRenamed()
+        {
+            _engine.DefineModule(@"const value1 = 1; const value2 = 2; export { value1 as renamed1, value2 as renamed2 }", "my-module");
```

Comment:
Modules don't seem to be imported when "defined", looks good, could be used to setup an engine with external optional dependencies. Is the module parsed at this point of on import? I would prefer `Set/Add/Create` module.